src/pages/index.tsx

```diff
@@ -1,109 +1,43 @@
-    submitRes();
-    // alert(
-    //   `Name = ${firstnameINPUT.toString()} and ${lastnameINPUT.toString()} , \n Course = ${course.toString()}`
-    // );
+export async function getServerSideProps(context: any) {
 
-    Swal.fire({
-      title: "Thank you for your submittion!",
-      text: `You have registered for ${firstnameINPUT.toString()} and ${lastnameINPUT.toString()} , \n Course = ${course.label.toString()}`,
-      // This app is intended to be used as a platform to calculate "who pays who what". This app was inspired by watching family members struggle to calculate the amount of $ owed after family vacations.
-      showClass: {
-        popup: "animate__animated animate__fadeInDown",
-      },
-      hideClass: {
-        popup: "animate__animated animate__fadeOutUp",
-      },
-    });
-  };
+  const res = await Promise.all([getLanguages(), getCourses()]);
+  const languages = res[0];
+  const courses = res[1];
 
-  //to submit a book into the database
-  const submitRes = async () => {
-    const response = await fetch("/api/makeRes", {
-      method: "POST",
-      body: JSON.stringify({
-        firstName: firstnameINPUT,
-        lastName: lastnameINPUT,
-        email: studentEmail,
-        language: language.label,
-        course: course,
-        middID: studentID,
-        resDate: dateValue,
-        type: "student",
-        is_cancelled: false,
-        on_waitlist: false,
-        attended: false,
-      }),
-      headers: {
-        "Content-Type": "application/json",
-      },
-    });
-    const data = await response.json();
-    console.log(data);
+  return {
+    props: {
+      languages,
+      courses,
+    } as HomeProps,
   };
+}
 
-  //function that checks the availability of the chosen date
-  const getAvailability = async () => {
-    console.log(language, dateValue);
-    const response = await fetch("/api/checkAvail", {
-      method: "POST",
-      body: JSON.stringify({
-        date: dateValue,
-        language: language.label,
-      }),
-      headers: {
-        "Content-Type": "application/json",
-      },
-    });
-    const data = await response.json().then((retuend) => {
-      console.log(retuend);
-      setavailabilitiy(retuend.data);
-    });
-  };
+/*
 
-  const defaultOption = languages[0];
 
-  useEffect(() => {
+TODO: signup should just be this file instead of in a separate component
 
-    //getAvailability();
-  }, [language, dateValue]);
+*/
+
+
+function Home({ languages, courses }: HomeProps) {
 
   return (
-    <div className={styles.container}>
-      <Head>
-        <title>LT Reservation</title>
-        {/* <link rel="icon" href="/favicon.ico" /> */}
-        {/* <link rel="icon" href="/favicon.svg" type="image/svg+xml" />
-        <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no" /> */}
-      </Head>
-      <Header />
-      <ReservationForm />
+    <div>
+      <Signup languages={languages} courses={courses} />
```

Comment:
There is no reason to separate Signup into its own component.  Just put the contents of signup into this return statement.